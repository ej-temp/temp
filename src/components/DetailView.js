import React from 'react';
import PropTypes from 'prop-types';

const DetailView = ({ name, temps }) => {
  const tempsList = temps.map(temp => (
    <li key={`${temp._id}`}>{temp.temp}</li>
  ));
  return (
    <>
      <h1>{name}</h1>
      <ul>{tempsList}</ul>
    </>
  );
};

DetailView.propTypes = {
  name: PropTypes.string.isRequired,
  temps: PropTypes.array.isRequired
};

export default DetailView;
