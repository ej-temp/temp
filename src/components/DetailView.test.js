import React from 'react';
import { shallow } from 'enzyme';
import DetailView from './DetailView';

describe('DetailView component', () => {
  it('renders DetailView', () => {
    const wrapper = shallow(<DetailView 
      name={'name'}
      temps={[]}
    />);
    expect(wrapper).toMatchSnapshot();
  });
});
