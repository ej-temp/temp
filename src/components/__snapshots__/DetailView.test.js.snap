// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailView component renders DetailView 1`] = `
<Fragment>
  <h1>
    name
  </h1>
  <ul />
</Fragment>
`;
